src/main/java/pedido/Pedido.java

```diff
@@ -27,19 +33,74 @@ public Cliente getCliente(){
     }
 
     public double calcularTotal(Cardapio cardapio){
-        double total= 0;
-        //TODO
+        double total = 0;
+
+        for (ItemPedido itemPedido : itens){
+            Base base = itemPedido.getShake().getBase();
+            Double preco = cardapio.buscarPreco(base);
+            TipoTamanho tamanho = itemPedido.getShake().getTipoTamanho();
+
+            if(tamanho == TipoTamanho.P) {
+                total += preco * itemPedido.getQuantidade();
+            } else if (tamanho == TipoTamanho.M) {
+                total += preco * 1.3 * itemPedido.getQuantidade();
+            } else {
+                total += preco * 1.5 * itemPedido.getQuantidade();
+            }
+        }
+
+        total += this.calcularAdicionais(cardapio);
         return total;
     }
 
+    private double calcularAdicionais(Cardapio cardapio) {
+        double precoTotalAdicionais = 0;
+        for(ItemPedido itemPedido : itens ) {
+            double precoAdicionais = 0;
+            List<Adicional> adicionais = itemPedido.getShake().getAdicionais();
+
+            for(Adicional adicional : adicionais) {
+                precoAdicionais += cardapio.buscarPreco(adicional);
+            }
+
+            precoTotalAdicionais += precoAdicionais * itemPedido.getQuantidade();
+        }
+
+        return precoTotalAdicionais;
+    }
+
+    int quantidade = 0;
+    int index = 0;
+
+    public boolean existeItem(ItemPedido itemPedido) {
+        for(ItemPedido item: itens) {
+            if (item.getShake().toString().equals(itemPedido.getShake().toString())) {
+                quantidade = item.getQuantidade();
+                index = itens.indexOf(item);
+                return true;
+            } else {
+                return false;
+            }
+        }
+        return false;
+    }
+
     public void adicionarItemPedido(ItemPedido itemPedidoAdicionado){
-        //TODO
+        if(existeItem(itemPedidoAdicionado)) {
+            itemPedidoAdicionado.setQuantidade(quantidade + itemPedidoAdicionado.getQuantidade());
+            itens.set(index, itemPedidoAdicionado);
+        } else {
+            itens.add(itemPedidoAdicionado);
+        }
     }
 
     public boolean removeItemPedido(ItemPedido itemPedidoRemovido) {
-        //substitua o true por uma condição
-        if (true) {
-            //TODO
+        if(existeItem(itemPedidoRemovido)) {
+            itemPedidoRemovido.setQuantidade(quantidade - 1);
+            itens.set(index, itemPedidoRemovido);
+            if(quantidade - 1 <= 0) {
+                itens.remove(itemPedidoRemovido);
+            }
         } else {
             throw new IllegalArgumentException("Item nao existe no pedido.");
         }
```

Comment:
`    public void removeItemPedido(final ItemPedido itemPedidoRemovido) throws ItemNotFound {
        var shakeIndex = itens.indexOf(itemPedidoRemovido);

        if (shakeIndex != -1) {
            var shake = itens.get(shakeIndex);
            if (shake.getQuantidade() == 1)
                itens.remove(itemPedidoRemovido);
            else
                shake.setQuantidade(shake.getQuantidade() - 1);
            return;
        }
        throw new ItemNotFound();
    }`

Reply:
o método abaixo já faz essa checagem por indexOf(). Eu tinha tentado fazer dessa forma que sugerida da primeira vez, mas alguns testes estavam dando erro. Optei por manter separada essa checagem pois ela também será usada no adicionarItemPedido. Mas entendo a sugestão, obrigada!

public boolean existeItem(ItemPedido itemPedido) {
        for(ItemPedido item: itens) {
            if (item.getShake().toString().equals(itemPedido.getShake().toString())) {
                quantidade = item.getQuantidade();
                index = itens.indexOf(item);
                return true;
            } else {
                return false;
            }
        }
        return false;
    }